lib/grandpa/message_handler.go

```diff
@@ -104,6 +104,11 @@ func (h *MessageHandler) handleNeighbourMessage(msg *NeighbourMessage) error {
 func (h *MessageHandler) handleCommitMessage(msg *CommitMessage) error {
 	logger.Debugf("received commit message, msg: %+v", msg)
 
+	err := verifyBlockHashAgainstBlockNumber(h.blockState, msg.Vote.Hash, uint(msg.Vote.Number))
+	if err != nil {
+		return err
```

Comment:
I have deliberately not wrapped the error `block hash does not correspond to given block number` is already enough. `failed verifying block hash against block number` does not provide more information here. `verifyBlockHashAgainstBlockNumber` is a tiny helping function rather than a task. If it was say a bigger task where context of the error was necessary, it would have made sense to wrap the error.

Reply:
Makes sense, thanks! :+1: